backend/src/org/commcare/resources/model/ResourceTable.java

```diff
@@ -446,15 +549,18 @@ public boolean upgradeTable(ResourceTable incoming) throws UnresolvedResourceExc
             return false;
         }
 
-        // Everything incoming should be marked either ready or upgrade. Upgrade elements
-        // should result in their counterpart in this table being unstaged (which can be
-        // reverted).
+        // Everything incoming should be marked either ready or upgrade.
+        // Upgrade elements should result in their counterpart in this table
+        // being unstaged (which can be reverted).
 
-        Stack<Resource> resources = incoming.GetResourceStack();
+        Stack<Resource> resources = incoming.getResourceStack();
         while (!resources.isEmpty()) {
             Resource r = resources.pop();
             Resource peer = this.getResourceWithId(r.getResourceId());
             if (peer == null) {
+                // no corresponding resource in this table; use incoming
+                // XXX PLM: Why is this needed? Only ever called on global
+                // table, which is thrown away and replaced by incoming table
                 this.addResource(r, Resource.RESOURCE_STATUS_INSTALLED);
```

Comment:
hm, agreed this seems incorrect. I think it might be the way that we ensure the tables are at parity when upgrading, but not sure.